libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -684,6 +744,130 @@ bool AC_Fence::sys_status_failed() const
     return false;
 }
 
+/*
+  return altitude max in desired frame
+ */
+bool AC_Fence::get_alt_frame_loc(const Location &loc, Location::AltFrame desired_frame, float alt, float &retalt) const
+{
+    // we need a location if we are handling terrain. For how the fence altitudes are used
+    // the best to use is our current location
+    Location::AltFrame alt_frame = (Location::AltFrame)_alt_frame.get();
+    if (alt_frame == desired_frame) {
+        retalt = alt;
+        return true;
+    }
+
+    Location loc2 = loc;
+    loc2.set_alt_cm(alt*100, alt_frame);
+
+    int32_t alt_cm;
+    if (!loc2.get_alt_cm(desired_frame, alt_cm)) {
+        return false;
+    }
+    retalt = alt_cm * 0.01;
+    return true;
+}
+
+/*
+  return altitude max in desired frame at current location
+ */
+bool AC_Fence::get_alt_frame(Location::AltFrame desired_frame, float alt, float &retalt) const
+{
+    Location::AltFrame alt_frame = (Location::AltFrame)_alt_frame.get();
+    if (alt_frame == desired_frame) {
+        retalt = alt;
+        return true;
+    }
+
+    // we need a location if we are handling terrain. For how the fence altitudes are used
+    // the best to use is our current location
+    Location loc;
+    if (!AP::ahrs().get_position(loc) &&
+        (desired_frame == Location::AltFrame::ABOVE_TERRAIN ||
+         alt_frame == Location::AltFrame::ABOVE_TERRAIN)) {
+        return false;
+    }
+
+    loc.set_alt_cm(alt*100, alt_frame);
+    int32_t alt_cm;
+    if (!loc.get_alt_cm(desired_frame, alt_cm)) {
+        return false;
+    }
+    retalt = alt_cm * 0.01;
+    return true;
+}
+
+/*
+  return altitude max in desired frame
+ */
+bool AC_Fence::get_alt_max(Location::AltFrame desired_frame, float &alt) const
+{
+    return get_alt_frame(desired_frame, _alt_max, alt);
+}
+
+/*
+  return altitude max in desired frame
+ */
+bool AC_Fence::get_alt_min(Location::AltFrame desired_frame, float &alt) const
+{
+    return get_alt_frame(desired_frame, _alt_min, alt);
+}
+
+/*
+  return altitude max in desired frame
+ */
+bool AC_Fence::get_alt_max_loc(const Location &loc, Location::AltFrame desired_frame, float &alt) const
```

Comment:
```suggestion
bool AC_Fence::get_alt_max(const Location &loc, Location::AltFrame desired_frame, float &alt) const
```

similarly elsewhere.  We don't need to encode the types of parameters taken by every method call in the method name.